components/bt/common/btc/profile/esp/blufi/nimble_host/esp_blufi.c

```diff
@@ -0,0 +1,476 @@
+#include <stdint.h>
+#include <string.h>
+#include <stdio.h>
+#include <assert.h>
+
+#include "btc_blufi_prf.h"
+#include "blufi_int.h"
+#include "esp_log.h"
+#include "esp_blufi_api.h"
+#include "esp_err.h"
+#include "btc/btc_task.h"
+#include "esp_blufi.h"
+#include "osi/allocator.h"   
+#include "console/console.h"
+
+/*nimBLE Host*/
+#include "esp_nimble_hci.h"
+#include "nimble/nimble_port.h"
+#include "nimble/nimble_port_freertos.h"
+#include "host/ble_hs.h"
+#include "host/util/util.h"
+#include "host/ble_uuid.h"
+#include "host/ble_gatt.h"
+
+#include "services/gap/ble_svc_gap.h"
+#include "services/gatt/ble_svc_gatt.h"
+
+#if (BLUFI_INCLUDED == TRUE)
+
+static uint8_t own_addr_type;
+static uint16_t  conn_handle;
+
+struct gatt_value gatt_values[SERVER_MAX_VALUES];
```

Comment:
Could you please help to explain what is the use of this extra `gatt_values` array ? Apart from that the value `SERVER_MAX_VALUES` should be limited to 3, your thoughts ?

Reply:
gatt_values will store information regarding that particular characteristics so that we can use them explicitly. e.g, in function esp_blufi_send_notify() , gatt_values[1].val_handle to handle notify operation on characteristic e2p.

Reply:
Yes, SERVER_MAX_VALUES can be limited to 3!👍